Adds CTI subscription navigation control component

yenienserrano opened the pull request:
### Description

Creates a reusable header navigation control system that dynamically positions CTI subscription components based on the new home page setting.

Introduces a complete CTI subscription workflow with modal dialogs for registration and status checking, supporting both active and pending subscription states.

Registers the navigation control in the main plugin to display CTI subscription options in either the top-right or bottom-left position depending on UI configuration.
 
### Evidence

[Screencast from 07-05-26 08:58:32.webm](https://github.com/user-attachments/assets/fc800491-0e64-493b-9ba5-a7083c03d8e1)





## Test

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## UI

| Test | Chrome | Firefox | Safari |
| --- |  --- | --- | --- |
| The **Register** entry must be rendered when not registered. | :black_circle: | :black_circle: | :black_circle: |
| Clicking it opens **Wazuh XDR registration** with intro text; **Cancel** / **Register** are in the footer. | :black_circle: | :black_circle: | :black_circle: |
| Clicking modal **Register** opens a new window/tab with the activation URL and `user_code`. | :black_circle: | :black_circle: | :black_circle: |
| In parallel, the modal shows **Complete activation in Wazuh Cloud**, user code / activation link callout, **Checking activation status**, and **Close**. | :black_circle: | :black_circle: | :black_circle: |
| If registration completes, success title **Your Wazuh XDR registration is complete** (and nav health success). | :black_circle: | :black_circle: | :black_circle: |
| If registration fails or expires, failed/expired UX allows retrying **Register** when applicable. | :black_circle: | :black_circle: | :black_circle: |
| After success, **Register** is replaced by **Wazuh Cloud** status with green (success) health. | :black_circle: | :black_circle: | :black_circle: |
| If modal is closed while pending, **Wazuh Cloud** status shows spinner and yellow (warning) health. | :black_circle: | :black_circle: | :black_circle: |
| Reopening modal while pending restores user code, activation link, and pending spinner state. | :black_circle: | :black_circle: | :black_circle: |

**Details**
<details>
<summary>:black_circle: The **Register** entry must be rendered when not registered.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Clicking it opens **Wazuh XDR registration** with intro text; **Cancel** / **Register** are in the footer.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Clicking modal **Register** opens a new window/tab with the activation URL and `user_code`.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: In parallel, the modal shows **Complete activation in Wazuh Cloud**, user code / activation link, **Checking activation status**, and **Close**.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: If registration completes, success title **Your Wazuh XDR registration is complete** (and nav health success).</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: If registration fails or expires, failed/expired UX allows retrying **Register** when applicable.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: After success, **Register** is replaced by **Wazuh Cloud** status with green (success) health.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: If modal is closed while pending, **Wazuh Cloud** status shows spinner and yellow (warning) health.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>

<details>
<summary>:black_circle: Reopening modal while pending restores user code, activation link, and pending spinner state.</summary>

Chrome - :black_circle:

Firefox - :black_circle:

Safari - :black_circle:

</details>


### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

yenienserrano committed:
Adds CTI subscription navigation control component

Creates a reusable header navigation control system that dynamically positions CTI subscription components based on the new home page setting.

Introduces a complete CTI subscription workflow with modal dialogs for registration and status checking, supporting both active and pending subscription states.

Registers the navigation control in the main plugin to display CTI subscription options in either the top-right or bottom-left position depending on UI configuration.

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Adds CTI subscription status integration with API

Implements real-time CTI subscription status checking by integrating with the Wazuh API instead of using mock data.

Creates new API endpoint to retrieve subscription information from cluster or manager nodes, automatically detecting cluster mode and querying the appropriate endpoint.

Refactors components to use async status checking and removes hardcoded status simulation, enabling dynamic status updates based on actual API responses.

Simplifies component props by removing unnecessary configuration parameters and centralizing status management logic.

yenienserrano committed:
Adds dynamic subscription status simulation for Wazuh API

Replaces static manager info responses with dynamic subscription status cycling through active, inactive, and processing states.

Implements stateful behavior using JavaScript controllers that rotate subscription status on each API call, enabling comprehensive testing of subscription state transitions in both manager and cluster node endpoints.

Removes outdated static response file and initializes subscription state as inactive for consistent test scenarios.

yenienserrano committed:
Refactors CTI subscription components and API integration

yenienserrano committed:
Add test

guidomodarelli commented on plugins/wazuh-check-updates/public/shared-components/cti-subscription/components/start-cti-subscription.test.tsx:
```diff
@@ -0,0 +1,91 @@
+      <StartCtiSubscription handleModalToggle={handleModalToggleMock} />,
+    );
+
+    expect(queryByText('Registry')).not.toBeInTheDocument();
```
Change to `Register`

guidomodarelli commented on plugins/wazuh-check-updates/public/shared-components/cti-subscription/components/start-cti-subscription.tsx:
```diff
@@ -0,0 +1,35 @@
+    <EuiToolTip position='bottom' content='Subscribe to CTI updates'>
+      <EuiButtonEmpty iconType='rocket' onClick={() => handleModalToggle()}>
+        <FormattedMessage
+          id='wazuhCheckUpdates.ctiSubscription.openModalRegistry'
+          defaultMessage='Registry'
```
```suggestion
          id='wazuhCheckUpdates.ctiSubscription.openModalRegister'
          defaultMessage='Register'
```

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Renames CTI subscription to registration across codebase, Adds polling mechanism to automatically check registration status updates and improves the user experience by removing manual status refresh requirement.

yenienserrano committed:
Fix imposter and labels

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Adds error handling for CTI registration failures

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Refactors CTI registration status system and terminology

Updates the CTI (Cyber Threat Intelligence) registration status flow to use more descriptive terminology and improves the authentication process workflow.

Renames status values from generic terms to specific CTI authentication states:
- 'inactive' becomes 'pending' for initial state
- 'active' becomes 'available' for successful registration
- Introduces 'polling' for ongoing registration process
- Introduces 'denied' for failed registration attempts

Enhances mock data structure by replacing 'registration' field with 'wazuh_cti_auth' to better reflect the authentication context and adds descriptive details for each status state.

Improves status checking logic to handle the new workflow states and provides clearer user feedback through updated UI components.

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Removes manager mock imposter files and improves CTI status handling

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Fixes subscription status handling and API response structure

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Replace getApiInfo to subscription

yenienserrano committed:
Add getStatusSubcriptionIndexerRoute

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Add comments

yenienserrano committed:
Refactor CTI registration components and update subscription handling logic

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Add connection to content manager

yenienserrano committed:
Improves CTI registration status handling and UI

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
revert imposter

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Implements CTI token API integration

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
Update CTI registration process with new verification URI and improved modal functionality

- Changed the verification URI in device_authorization.json to point to the correct activation page.
- Enhanced the ModalCti component to display the user code and activation link after registration.
- Updated the token retrieval process to use a client ID instead of current API ID.
- Added a new service to resolve the CTI console base URL for better configuration management.
- Improved test cases for ModalCti to ensure correct rendering and functionality.

felipegonzalezmv committed:
docs(docker): document CTI token-only dashboard flow and set CTI Console URL in osd-dev

Point osd-dev at the shared Imposter service and document that registration
state comes from the same OAuth device token endpoint, not subscription APIs.

Made-with: Cursor

felipegonzalezmv committed:
feat(wazuh-check-updates): proxy CTI OAuth device token poll and remove subscription routes

Extend POST /cti-registration/token to forward grant_type plus device_code to
the CTI Console, resolve client_id from OpenSearch cluster_uuid when omitted,
and delete the subscription proxy services and routes. Add route tests for the
token handler.

Made-with: Cursor

felipegonzalezmv committed:
feat(wazuh-check-updates): derive CTI registration UI from token poll and device flow

Replace the subscription HTTP client with cti-registration-status, persist
device_code for polling, consolidate the registration modal with device-auth
links, and remove the separate status modal.

Made-with: Cursor

felipegonzalezmv committed:
refactor(wazuh-check-updates): streamline CTI registration state management

Removed constants related to session storage for device code and registration status. Introduced a new service, , to manage the in-memory state of the CTI device authorization flow. Updated components and tests to utilize the new state management approach, enhancing clarity and maintainability.

felipegonzalezmv committed:
refactor(wazuh-check-updates): clean up CTI flow state and registration status documentation

Removed outdated comments and documentation from the CTI flow state and registration status files to enhance code clarity. This includes eliminating unnecessary comments related to session storage and device authorization, streamlining the codebase for better maintainability.

felipegonzalezmv committed:
feat(wazuh-check-updates): enhance CTI registration flow with improved polling and error handling

- Introduced constants for device polling intervals and expiration times to standardize behavior.
- Updated the CTI flow state management to handle device authorization expiry and polling intervals.
- Enhanced the registration status handling to reflect registration failures and provide user feedback.
- Improved the ModalCti component to display registration progress and success/failure messages.
- Refactored the useCtiStatus hook to manage polling based on device flow state and handle silent updates.
- Added tests to ensure correct behavior of the registration process and error handling.

This update aims to provide a more robust and user-friendly experience during the CTI registration process.

felipegonzalezmv committed:
feat(wazuh-check-updates): add CTI registration status API and enhance flow management

- Introduced a new API endpoint for fetching CTI registration status, providing real-time updates on the registration process.
- Updated the constants to include the new registration status route.
- Enhanced the CTI flow management by implementing a dedicated store for managing registration state, including in-progress and completed statuses.
- Refactored related components and hooks to utilize the new API and store, improving the overall user experience during registration.
- Added tests to ensure the correctness of the new registration status functionality.

This update aims to provide a more comprehensive and user-friendly CTI registration experience.

felipegonzalezmv committed:
feat(wazuh-check-updates): enhance CTI registration feedback and UI components

- Added a success status message for CTI registration, improving user feedback upon successful registration.
- Updated the fetchCtiRegistrationStatus function to return the new success message.
- Refactored the CtiRegistration component to utilize EuiFlexGroup for better layout management.
- Enhanced the ModalCti component to display a progress indicator during registration and show success details.
- Improved the StatusCtiRegistration component to handle background registration states and provide visual feedback.

This update aims to create a more informative and user-friendly experience during the CTI registration process.

felipegonzalezmv committed:
refactor(wazuh-check-updates): improve error messages for CTI console configuration

felipegonzalezmv committed:
feat(wazuh-check-updates): add Content Manager subscription mock and integration

- Implemented a mock for the Content Manager subscription API to handle access token registration.
- Updated OpenAPI specifications to include the new subscription endpoint and its request/response schema.
- Enhanced the README documentation to describe the mock's usage and security considerations.
- Integrated the subscription mock into the CTI registration flow, ensuring server-side handling of access tokens.
- Added tests to validate the new subscription functionality and error handling.

This update aims to facilitate the development and testing of the CTI registration process with a stand-in for the Content Manager API.

felipegonzalezmv committed:
feat(wazuh-check-updates): enhance CTI flow state management with device authorization links

felipegonzalezmv committed:
feat(wazuh-check-updates): enhance CTI registration with persisted credentials management

- Introduced functionality to check for persisted CTI credentials, improving the registration flow.
- Updated the CTI registration status API to include a flag indicating the presence of persisted credentials.
- Refactored related components and services to utilize the new credentials management, ensuring a seamless user experience.
- Enhanced tests to validate the new functionality and its integration within the CTI registration process.

This update aims to provide better handling of CTI credentials, improving the overall registration experience.

felipegonzalezmv committed:
refactor(wazuh-check-updates): improve CTI registration UI and loading states

- Updated the CTI registration modal to include a loading spinner while awaiting server snapshot readiness.
- Enhanced the display logic for registration states, ensuring a smoother user experience during the registration process.
- Refactored tests to utilize screen queries for better async handling and improved readability.
- Adjusted constants documentation for clarity regarding paths appended to the CTI console base URL.

This update aims to provide a more responsive and informative interface during CTI registration.

felipegonzalezmv committed:
feat(wazuh-check-updates): add configuration option for CTI registration UI

- Introduced a new configuration option  to control the visibility of the CTI registration UI.
- Updated the Wazuh plugin to conditionally register the header navigation control based on the new configuration.
- Enhanced the plugin initialization to accept the configuration context, improving flexibility and maintainability.

This update aims to provide better control over the CTI registration interface, allowing for more tailored user experiences.

felipegonzalezmv committed:
fix(wazuh-check-updates): update CTI registration UI and test assertions

- Changed the default message for device authorization links to provide clearer instructions for users.
- Removed the display of the complete verification URI in the UI for a cleaner look.
- Updated test assertions to reflect changes in the modal's activation status messaging and verification link handling.

This update aims to enhance user clarity during the CTI registration process and improve test accuracy.

felipegonzalezmv committed:
refactor(wazuh-check-updates): streamline CTI registration status fetching and visibility handling

- Removed the optional parameter for skipping hydration in the fetchCtiRegistrationStatus function, ensuring consistent state updates.
- Enhanced the useCtiStatus hook to automatically refetch status when the document becomes visible, improving responsiveness to user interactions.
- Updated comments for clarity regarding the hydration process and its impact on background polling.

This update aims to improve the reliability of CTI registration status updates and enhance user experience during visibility changes.

felipegonzalezmv committed:
refactor(wazuh-check-updates): rename persisted credentials flag to isRegistered

- Replaced the hasPersistedCtiCredentials flag with isRegistered across the codebase for clarity and consistency.
- Updated related functions and tests to reflect the new naming convention, ensuring accurate status checks during CTI registration.
- Enhanced the CTI registration status API to return the updated isRegistered flag.

This update aims to improve code readability and maintainability by standardizing the terminology used for CTI registration status.

felipegonzalezmv committed:
feat(wazuh-check-updates): implement subscription status retrieval for CTI registration

- Added a new GET endpoint to retrieve the subscription registration status, including the clientId as a required query parameter.
- Enhanced the OpenAPI specification to document the new endpoint and its response structure.
- Updated the subscription mock to handle GET requests and return the registration status.
- Refactored related components and services to utilize the new subscription status API, improving the overall registration flow.
- Added tests to validate the new functionality and ensure accurate status reporting.

This update aims to provide a clearer view of the CTI registration status, enhancing user experience and system reliability.

felipegonzalezmv committed:
feat(wazuh-check-updates): enhance device authorization handling in CTI registration modal

felipegonzalezmv committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
refactor(wazuh-check-updates): update verification URIs and improve test structure

- Replaced hardcoded verification URIs with example URLs across multiple test files to enhance clarity and maintainability.
- Added cleanup logic in the ModalCti component tests to ensure proper resource management after each test.
- Updated test assertions to utilize query selectors for improved readability and consistency.

This update aims to streamline the testing process and improve the overall structure of the CTI registration tests.

felipegonzalezmv committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
Merge remote-tracking branch 'origin/feature/2837-cti-paid-subscriptions---create-the-registration-process-ui'

felipegonzalezmv committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
Update subscription status in JSON response to reflect unregistered state

felipegonzalezmv committed:
refactor(cti-registration): update CTI registration text and button labels for clarity

felipegonzalezmv committed:
refactor(cti-registration): update activation terminology and improve loading state styling

felipegonzalezmv committed:
feat(cti-registration): add Wazuh Cloud portal link and update UI text for clarity

felipegonzalezmv committed:
refactor(cti-registration): remove mock Content Manager subscription endpoints and related files

- Deleted mock endpoints and responses for the Content Manager subscription from the configuration and OpenAPI specifications.
- Updated documentation to reflect that the Content Manager subscription is no longer mocked and should be accessed directly from the OpenSearch cluster.
- Refactored related code to remove dependencies on the now-removed mock implementation.

felipegonzalezmv committed:
-merge branch

felipegonzalezmv committed:
feat(docker): add Mailpit service for local email testing in development environment

felipegonzalezmv committed:
fix prettier

felipegonzalezmv committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
refactor(cti-registration): streamline CTI Console URL handling and update documentation

- Removed the hardcoded WAZUH_CTI_CONSOLE_BASE_URL from the Docker configuration.
- Updated constants to clarify the usage of the Wazuh Cloud CTI Console API base URL.
- Refactored the CTI Console URL retrieval to utilize the updated constant for improved maintainability.
- Enhanced documentation to provide clearer guidance on environment configuration for the CTI Console.

felipegonzalezmv committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv committed:
Merge remote-tracking branch 'origin/feature/2837-cti-paid-subscriptions---create-the-registration-process-ui'

yenienserrano committed:
Merge branch 'main' into feature/2837-cti-paid-subscriptions---create-the-registration-process-ui

felipegonzalezmv reviewed (approved):
LGTM 🟢
